File: puzzles/ts/there-is-no-spoon1/there-is-no-spoon1.ts
const EMPTY_CELL = '.';
const width: number = parseInt(readline()); // the number of cells on the X axis
const height: number = parseInt(readline()); // the number of cells on the Y axis
let grid: string[] = [];
for (let i: number = 0; i < height; i++) {
  const row: string = readline(); // width characters, each either 0 or .
  grid.push(row);
}

for (let y: number = 0; y < height; y++) {
  for (let x: number = 0; x < width; x++) {
    let output: string = "";
    if (grid[y][x] == EMPTY_CELL) {
      continue;
    }
    // current node
    output += `${x} ${y} `;

    // right neighbor
    let x2: number = -1;
    let y2: number = -1;
    for (let i: number = x + 1; i < width; i++) {
      if (grid[y][i] != EMPTY_CELL) {
        x2 = i;
        y2 = y;
        break
      }
    }
    output += `${x2} ${y2} `;

    // bottom neighbor
    let x3: number = -1;
    let y3: number = -1;
    for (let j: number = y + 1; j < height; j++) {
      if (grid[j][x] != EMPTY_CELL) {
        x3 = x;
        y3 = j;
        break
      }
    }
    output += `${x3} ${y3}`;

    // Three coordinates: a node, its right neighbor, its bottom neighbor
    console.log(output);
  }
}

File: puzzles/cs/there-is-no-spoon1/there-is-no-spoon1.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        const char EMPTY = '.';
        const char NODE = '0';
        List<string> grid = new List<string>();

        int width = int.Parse(Console.ReadLine()); // the number of cells on the X axis
        int height = int.Parse(Console.ReadLine()); // the number of cells on the Y axis
        for (int i = 0; i < height; i++)
        {
            grid.Add(Console.ReadLine());
        }

        for (int y = 0; y < height; y++)
        {
            for (int x = 0; x < width; x++)
            {
                if (grid[y][x] == NODE)
                {
                    // node
                    string output = $"{x} {y} ";

                    // right neighbor
                    char neighbor = EMPTY;
                    for (int neighbor_x = x + 1; neighbor_x < width; neighbor_x++)
                    {
                        neighbor = grid[y][neighbor_x];
                        if (neighbor == NODE)
                        {
                            output += $"{neighbor_x} {y} ";
                            break;
                        }
                    }

                    if (neighbor == EMPTY)
                    {
                        output += "-1 -1 ";
                    }

                    // bottom neighbor
                    neighbor = EMPTY;
                    for (int neighbor_y = y + 1; neighbor_y < height; neighbor_y++)
                    {
                        neighbor = grid[neighbor_y][x];
                        if (neighbor == NODE)
                        {
                            output += $"{x} {neighbor_y} ";
                            break;
                        }
                    }

                    if (neighbor == EMPTY)
                    {
                        output += "-1 -1 ";
                    }

                    Console.WriteLine(output);
                }
            }
        }
    }
}
